nix/lib.nix

```diff
@@ -0,0 +1,33 @@
+{ pkgs, lib ? pkgs.lib }: {
+  # Create an attrsOf module wrapper (`services.${name}`) around the given submodule.
+  #
+  # where module filename is of form `${name}.nix`. The submodule takes this
+  # 'name' parameter, and is expected to set the final process-compose config in
+  # its `outputs.settings` option.
+  multiService = mod:
+    let
+      # Derive name from filename
+      name = lib.pipe mod [
+        builtins.baseNameOf
+        (lib.strings.splitString ".")
+        builtins.head
+      ];
+    in
+    { config, ... }: {
```

Comment:
Ya, let's go with this approach. 

But I don't believe you need the `{ lib }` parameter either (you can use `pkgs.lib` instead and move the `let` block inside the module block). Which means this could simply be exposed as `lib.multiService`.

Reply:
`pkgs.lib` does require configuring and evaluating `pkgs`, so in order to have a more consistent pattern across the ecosystem, it is beneficial to accept `lib` separately. If it doesn't matter in your case, take `lib ? pkgs.lib`.

Reply:
> Btw. is the lib flake output expected to be `{ lib } -> { ... }` or just `{ ... }`?

It doesn't come with expectations, but you could do `{ ... }` if you take `lib` from the flake inputs.
Here flakes is an example of a framework that does too much, even if it's just fairly simple dependency injection. It's there, so we're expected to use it, even if it's not quite the right thing. Maybe it should be `{ ... } // { withLib = ... }` if you want to both satisfy all expectations and make the interface fully general.
Could be over-engineered or just right. I'll leave that up to you.

Reply:
> `pkgs.lib` does require configuring and evaluating `pkgs`, so in order to have a more consistent pattern across the ecosystem, it is beneficial to accept `lib` separately.

makes sense